astronomer/providers/google/cloud/xcom_backends/gcs.py

```diff
@@ -0,0 +1,81 @@
+import json
+import os
+import pickle  # nosec
+import uuid
+from datetime import date, datetime
+from typing import Any
+
+import pandas as pd
+from airflow.configuration import conf
+from airflow.models.xcom import BaseXCom
+from airflow.providers.google.cloud.hooks.gcs import GCSHook
+
+
+class GCSXComBackend(BaseXCom):
+    """
+    Custom XCom persistence class extends base to support various datatypes.
+    To use this XCom Backend, add the environment variable `AIRFLOW__CORE__XCOM_BACKEND`
+    to your environment and set it to
+    `astronomer.providers.google.cloud.xcom_backends.gcs.GCSXComBackend`
+    """
+
+    PREFIX = os.getenv("XCOM_BACKEND_PREFIX", "gcs_xcom_")
+    GCP_CONN_ID = os.getenv("XCOM_BACKEND_CONNECTION_NAME", "google_cloud_default")
+    BUCKET_NAME = os.getenv("XCOM_BACKEND_BUCKET_NAME", "airflow_xcom_backend_default_bucket")
+    PANDAS_DATAFRAME = "dataframe"
+    DATETIME_OBJECT = "datetime"
+
+    @staticmethod
+    def write_and_upload_value(value: Any) -> str:
+        """Convert to string and upload to GCS"""
+        key_str = GCSXComBackend.PREFIX + str(uuid.uuid4())
```

Comment:
```suggestion
        key_str = f"{GCSXComBackend.PREFIX}{uuid.uuid4()}"
```

Same for other `+` lines below. This is more performant.